mantidproject/mantid
# GetEI fails on noisy functions

**Description of work.**
Additional logic has been added to the GetEi2 algorithm to ensure that a divide-by-zero error doesn't occur in rare circumstances when a local maxium on the side of a peak meets cirtain criteria.

**Report to:** [Ross Stewart]

**To test:**
Run the sample script in the linked issue and ensure Ei is calculated with no errors are thrown. Unit tests have been added to recreate the bug using a toy dataset.


Fixes #31429. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
